sdk/python/kfp/cli/utils/deprecated_alias_group.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2022 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, List, Tuple, Union
+
+import click
+
+
+def deprecated_alias_group_factory(deprecated_map: Dict[str, str]):
+    """Closure that returns a class that implements the deprecated alias group.
+
+    Args:
+        deprecated_map (Dict[str, str]): Dictionary mapping old deprecated names to new names.
+
+
+    Returns:
+        DeprecatedAliasGroup: A class that implements the deprecated alias group.
```

Comment:
nit: type annotate this in the function signature?

Reply:
I'm not sure if this is possible. My attempts:

1. This doesn't work, because `DeprecatedAliasGroup` is not defined yet:
```python
def deprecated_alias_group_factory(...) -> DeprecatedAliasGroup:
```

2. And this doesn't work, because mypy does not support this type for forward ref (mypy error: Name "DeprecatedAliasGroup" is not defined):

```python
def deprecated_alias_group_factory(...) -> 'DeprecatedAliasGroup':
```

I opted for omission over using `Any`. WDYT?

Reply:
> `def deprecated_alias_group_factory(...) -> 'DeprecatedAliasGroup':`

We have been using this forward declaration pattern in our code base. Now I do recall that mypy would complain on such annotation, but we weren't enforcing the mypy check in the past.
One way to make it "work" is to disable the mypy error on this specific line. Then the annotation might be slightly more helpful for devs looking at this code. 
It's very minor. Feel free to ignore my original comment. :)

Reply:
I like that approach 👍🏻

Reply:
Updated (and also fixed one other mypy error)